lib/vector-buffers/src/variants/disk_v2/tests/size_limits.rs

```diff
@@ -377,44 +393,33 @@ async fn writer_try_write_returns_when_buffer_is_full() {
         let data_dir = dir.to_path_buf();
 
         async move {
-            // Create our buffer with an arbitrarily low max buffer size, and two write sizes that
-            // will both fit just under the limit but will provide no chance for another write to
-            // fit.
-            //
-            // The sizes are different so that we can assert that we got back the expected record at
-            // each read we perform.
+            // Create our buffer with an arbitrarily low max buffer size, such that we can construct
+            // two writes that would otherwise fit by themselves but will end up with the second not
+            // being able to fit as the buffer is exactly full.
+            let write_size = 96;
```

Comment:
Might be interesting to let proptest vary this write_size as well.

Reply:
This ends up happening already in the main disk v2 model, where we vary all configuration parameters (max buffer size, max data file size, and max record size, and write buffer size) as well as the size of records themselves.